docs/src/adr/04adr-reactor-stub.md

```diff
@@ -0,0 +1,130 @@
+# ADR-04: Atomkraft reactor stub command
+
+## Context
+This ADR describes one component of Atomkraft, as decribed in [ADR-01](01adr-principles-architecture.md).
+
+## Status
+This is an early version of the ADR, meant for the first prototype.
+Major changes to the scope are very likely.
+
+## Reactor Stub
+
+Once the user has a trace (either written by hand, or obtained from the system model),  they need to write a *reactor*.
+A reactor is a set of Python functions connecting the actions of the trace to executions of the code.
+
+In this ADR we describe a command `atomkraft reactor ...` which creates a stub of the reactor,
+auto-generating all the boilerplate, and leaving it to the user to only fill in the application-specific parts.
+
+## Command Line Interface
+
+The interface of the command looks like this:
+`atomkraft reactor <actions_list> <variables_list> [<reactor-stub-file>]`
+where
+ - `actions_list` is a list of actions for which to generate stubs
+ - `variables_list` is a list of all relevant state variables (those that could appear in traces of interest)
+ - `reactor_stub_file` is a path at which the reactor file should be created.
+ If omitted, a default path is used: `atomkraft/reactors/reactor.py`
+
+## Implementation
+
+ The `Reactor` class implements the desired behavior.
+ Its member functions are:
+  - `generate_reactor(actions_list, variables_list, stub_file_path=None) -> Path`:
+  generates the stub, containing one `@step` function for each action from the `action_list`,
+  and each of these functions takes variables from the `variables_list` as arguments.
+  The reactor stub is created at `stub_file_path` name location
+  (if it is `None`, a default location is used).
+  The function returns the path to the generated reactor stub and updates the value of `reactor` under `atomkraft.toml`.
+
+  - `check_reactor(trace, keypath="action", reactor=None) -> bool`: it checks if the `reactor` (default reactor if `None`) is compatible with the `trace`. A reactor is compatible with a trace if each action appearing in `trace` is matched with a function in `reactor`. The `keypath` argument defines which field of the ITF trace contains the action (if omitted, it takes the default value "action"). The function returns `True` if the check is successful, and `False` otherwise.
+
+  - `get_reactor() -> Path`: returns the path to the reactor from `atomkraft.toml`
+
+
+### Stub content
+
+ The stub should include:
+
+
+  - a stub for the `chain_testnet` initialization function
+  ```python
+  @pytest.fixture(scope="session")
```

Comment:
@rnbguy , is it fair to assume that I'll get all these values  (`node_config`, `genesis_config`) from the config file?